homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -75,22 +90,36 @@ def __init__(self, name, device, model, unique_id):
         self._device = device
         self._model = model
         self._unique_id = unique_id
-
         self._icon = "mdi:cloud"
-        self._unit_of_measurement = "AQI"
         self._available = None
         self._state = None
-        self._state_attrs = {
-            ATTR_POWER: None,
-            ATTR_BATTERY_LEVEL: None,
-            ATTR_CHARGING: None,
-            ATTR_DISPLAY_CLOCK: None,
-            ATTR_NIGHT_MODE: None,
-            ATTR_NIGHT_TIME_BEGIN: None,
-            ATTR_NIGHT_TIME_END: None,
-            ATTR_SENSOR_STATE: None,
-            ATTR_MODEL: self._model,
-        }
+
+        if (
+            self._model == MODEL_XIAOMI_AIRQUALITYMONITOR_S1
+            or self._model == MODEL_XIAOMI_AIRQUALITYMONITOR_B1
+        ):
+            self._unit_of_measurement = "°C"
+            self._state_attrs = {
+                ATTR_BATTERY_LEVEL: None,
+                ATTR_CO2: None,
+                ATTR_HUMIDITY: None,
```

Comment:
The humidity measurement should be its own entity with `device_class` humidity and not a state attribute.